api.rb

```diff
@@ -0,0 +1,24 @@
+require 'sinatra'
+require 'jwt'
+require_relative 'lib/notify'
+
+class MyApi < Sinatra::Base
+  before do
+    halt 401 if request.env['HTTP_AUTHORIZATION'].nil?
+    acces_token = request.env['HTTP_AUTHORIZATION'].slice(7..-1)
+    JWT.decode acces_token, public_key, true, { algorithm: 'RS256' }
+  rescue StandardError
+    halt 403
+  end
+
+  post '/deploy-in-progress' do
+    halt 404 if params['env'].nil?
```

Comment:
To make it more explicit what `env` stands for:

```suggestion
    halt 404 if params['target_environment'].nil?
```